es-core/src/AudioManager.cpp

```diff
@@ -135,21 +110,175 @@ void AudioManager::unregisterSound(std::shared_ptr<Sound> & sound)
 void AudioManager::play()
 {
 	getInstance();
-
-	//unpause audio, the mixer will figure out if samples need to be played...
-	SDL_PauseAudio(0);
 }
 
 void AudioManager::stop()
 {
 	//stop playing all Sounds
 	for(unsigned int i = 0; i < sSoundVector.size(); i++)
-	{
 		if(sSoundVector.at(i)->isPlaying())
-		{
 			sSoundVector[i]->stop();
+}
+
+void AudioManager::getMusicIn(const std::string &path, std::vector<std::string>& all_matching_files)
+{	
+    if(!Utils::FileSystem::isDirectory(path)) {
+		return;
+    }
+    auto dirContent = Utils::FileSystem::getDirContent(path);
+    for(auto it = dirContent.cbegin(); it != dirContent.cend(); ++it)
+	{	
+		if(Utils::FileSystem::isDirectory(*it))
+		{
+			if(*it == "." || *it == "..")
+				getMusicIn(*it, all_matching_files);
+					
+		}
+		else
+		{
+			std::string extension = Utils::String::toLower(Utils::FileSystem::getExtension(*it));
+			if(extension == ".mp3" || extension == ".ogg")
+				all_matching_files.push_back(*it);
 		}
+	}		
+}
+
+void AudioManager::playRandomMusic(bool continueIfPlaying)
+{
+	if(!Settings::getInstance()->getBool("EnableMusic"))
+		return;
+
+	std::vector<std::string> musics;
+	
+	// check in RetroPie music directory
+	if(musics.empty())
+		getMusicIn(Utils::FileSystem::getHomePath() + "/RetroPie/roms/music", musics);
+  
+	// check in system sound directory
+	if(musics.empty())
```

Comment:
Is this an actual RetroPie folder? Wouldn't it probably something closer to /opt/RetroPie/music? Or is it a non-pi specific folder?

Reply:
Audio files can be placed in $HOME/RetroPie/roms/music, /usr/share/RetroPie/music, or /.emulationstation/music.

Reply:
Correct. What I'm saying is that "/usr/share/XXX" is a Batocera path :) RetroPie doesn't create that folder by default - the path you're probably looking for here is "/opt/retropie/music" then. @joolswills thoughts?

Reply:
Doesn't this mean that if there is an mp3 or ogg in "$HOME/RetroPie/roms/music" it will ignore any files in the other directories?